app/models/upload.rb

```diff
@@ -234,7 +234,7 @@ def self.migrate_to_new_scheme(limit = nil)
           FileHelper.optimize_image!(path) if FileHelper.is_supported_image?(File.basename(path))
           # store to new location & update the filesize
           File.open(path) do |f|
-            upload.url = Discourse.store.store_upload(f, upload)
+            upload.url, upload.etag = Discourse.store.store_upload(f, upload)
```

Comment:
I'm curious what happens in this assignment when `Discourse.store` is a different store? The pattern that we're using now requires all new devs to be aware of the fact that only the s3 store will return an etag.